### PR TITLE
feat(checkbox): Add disabled state color mixins

### LINKED ISSUE #4774: [Checkbox] Add mixins/ink-color screenshot test for checkbox
#4773 removes flaky test `spec/mdc-checkbox/mixins/ink-color.html`. We need to add this test back by trying out different color combination or other feasible solution.

### PR DESCRIPTION
**Pull request by allan-chen**
Adds the `mdc-checkbox-disabled-container-colors` and `mdc-checkbox-disabled-ink-color` mixins to allow customizing disabled state colors. 

Screenshot test for ink colors that was previously removed has been re-added. (fixes #4774)

BREAKING CHANGE: `mdc-checkbox-ink-color` mixin now only applies to enabled checkboxes

### CODE REVIEW

File: packages/mdc-checkbox/_mixins.scss [hunk 1]
```diff
@@ -324,18 +322,77 @@ $mdc-checkbox-ripple-target: ".mdc-checkbox__ripple";
...
 
-@mixin mdc-checkbox-ink-color($color, $query: mdc-feature-all()) {
+///
+/// Customizes disabled state stroke and fill colors
+/// @param {Color} $unmarked-stroke-color - The desired stroke color for the unmarked state
+/// @param {Color} $unmarked-fill-color - The desired fill color for the unmarked state
+/// @param {Color} $marked-stroke-color - The desired stroke color for the marked state
+/// @param {Color} $marked-fill-color - The desired fill color for the marked state
+///
+@mixin mdc-checkbox-disabled-container-colors(
```
Comment (abhiomkar) [inline packages/mdc-checkbox/_mixins.scss]: As discussed, it's worth investigating to create disabled selector mixin.

See example here:
https://www.sassmeister.com/gist/9168132c52a31cd48917e64beb9e5a63

Comment (allan-chen) [reply packages/mdc-checkbox/_mixins.scss]: Separated core styles from selector logic in both this change (checkbox) and radio, as later discussed.

File: packages/mdc-checkbox/_mixins.scss [hunk 2]
```diff
@@ -685,6 +716,22 @@ $mdc-checkbox-ripple-target: ".mdc-checkbox__ripple";
   }
 }
 
+@mixin mdc-checkbox-ink-color_($color, $query: mdc-feature-all()) {
```
[the remark this answers is not in the document]
Comment (allan-chen) [reply packages/mdc-checkbox/_mixins.scss]: Done

File: packages/mdc-checkbox/_mixins.scss [hunk 3]
```diff
@@ -504,6 +514,18 @@ $mdc-checkbox-ripple-target: ".mdc-checkbox__ripple";
 
 // Background
 
+@mixin mdc-checkbox-background-selector-enabled_ {
```
[the remark this answers is not in the document]
Comment (allan-chen) [reply packages/mdc-checkbox/_mixins.scss]: Made changes according to these guidelines

File: packages/mdc-checkbox/_mixins.scss [hunk 4]
```diff
@@ -259,6 +257,15 @@ $mdc-checkbox-ripple-target: ".mdc-checkbox__ripple";
...
 }
 
+///
+/// Sets stroke & fill colors for both marked and unmarked state of enabled checkbox.
+/// Set $generate-keyframes to false to prevent the mixin from generating @keyframes
+/// @param {Color} $unmarked-stroke-color - The desired stroke color for the unmarked state
+/// @param {Color} $unmarked-fill-color - The desired fill color for the unmarked state
+/// @param {Color} $marked-stroke-color - The desired stroke color for the marked state
+/// @param {Color} $marked-fill-color - The desired fill color for the marked state
+/// @param {bool} $generate-keyframes [true] - Whether animation keyframes should be generated
```
Comment (abhiomkar) [inline packages/mdc-checkbox/_mixins.scss]: nit: `{Boolean}`

**Pull request merged**